spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java

```diff
@@ -125,7 +119,7 @@ protected void onInit() {
 	}
 
 	@Override
-	public void destroy() throws Exception {
+	public void destroy() {
```

Comment:
We can't do this - breaking change for anyone that has subclassed and implemented `destroy()`.

I suggest we use `// NOSONAR TODO: remove throws in 5.2`, which removing it from our subclasses.

Reply:
OK. 
Do we remove from our subclasses now as I did in this PR?

Sorry, if I misunderstood you...